### PR TITLE
fix: there is some a little bug,maybe it's helpful

### PR DESCRIPTION
**Pull request by csh0101**
in the bitarray.go,the func String() ' receiver should be a BitArray  pointer? i guess that you write the func BitArray'String() maybe forget lost the *,if not, i feel sorry..

### CODE REVIEW

File: bitarray_internal_test.go [hunk 1]
```diff
@@ -31,9 +31,6 @@ func (ba *BitArray) V() {
 	case ba.b != nil && len(ba.b) == 0:
 		panicf("V: ba.b is an empty slice, must be nil: %08b", ba.b)
 
-	case ba.b == nil:
```
Comment (csh0101) [inline bitarray_internal_test.go]: it seems my eyes is not goodful。 😄

**Comment by tunabay** [comment]
Thank you for PR, but it's intentional, not a bug.
I chose a value receiver so that both `BitArray` and `*BitArray` types implement `fmt.Stringer`. Also, `BitArray` is small enough that a value receiver would be harmless.

Try this and switch the receiver type of `X`: https://play.golang.org/p/Af7Zv5pYzub

**Comment by csh0101** [comment]
thanks，i think i m study something fron this

**Comment by tunabay** [comment]
Your fix may be good for consistency. There are various opinions about which type of receiver is better, but for now I decide not to fix this and close. Thanks.

**Pull request closed without merging**